annet/adapters/netbox/models.py

```diff
@@ -0,0 +1,155 @@
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import List, Optional, Any
+
+from annet.annlib.netdev.views.hardware import HardwareView
+
+
+@dataclass
+class Entity:
+    id: int
+    name: str
+
+
+@dataclass
+class Label:
+    value: str
+    label: str
+
+
+@dataclass
+class IpFamily:
+    value: int
+    label: str
+
+
+@dataclass
+class IpAddress:
+    id: int
+    assigned_object_id: int
+    display: str
+    family: IpFamily
+    address: str
+    status: Label
+    tags: List[Entity]
+    created: datetime
+    last_updated: datetime
+
+
+@dataclass
+class Interface(Entity):
+    device: Entity
+    enabled: bool
+    display: str = ""  # added in 3.x
+
+    # filled later
+    ip_addreses: List[IpAddress] = field(default_factory=list)
+
+
+@dataclass
+class DeviceType:
+    id: int
+    manufacturer: Entity
+    model: str
+
+
+@dataclass
+class DeviceIp:
+    id: int
+    display: str
+    address: str
+    family: int
+
+
+@dataclass
+class Device(Entity):
+    display: str  # renamed in 3.x from display_name
+    device_type: DeviceType
+    device_role: Entity
+    tenant: Optional[Entity]
+    platform: Optional[Entity]
+    serial: str
+    asset_tag: Optional[str]
+    site: Entity
+    rack: Optional[Entity]
+    position: Optional[float]
+    face: Optional[Label]
+    status: Label
+    primary_ip: Optional[DeviceIp]
+    primary_ip4: Optional[DeviceIp]
+    primary_ip6: Optional[DeviceIp]
+    tags: List[Entity]
+    custom_fields: dict[str, Any]
+    created: datetime
+    last_updated: datetime
+    # filled later
+    interfaces: List[Interface] = field(default_factory=list)
+
+    # compat
+    def __hash__(self):
+        return hash(self.id)
+
+    @property
+    def fqdn(self):
+        return self.name
+
+    @property
+    def hostname(self):
+        return self.name
+
+    def is_pc(self):
```

Comment:
Could you please split this into separate classes. It would be reflecting the following enities:

- the dto of the netbox object itself
- the implementation of annet.storage.Device interface
- the object that combines the two

It can be done in the following way (names could be better tho, also the other approaches is possible):

```
@dataclass
class NeboxDTO:
   display: str
   ...

class NetboxAnnetDeviceImpl:
   def __init__(self, netbox_dto):
     self._netbox_dto = netbox_dto
  def fqdn(self):
     return self._netbox_dto.name
  ...

 class NetboxDevice(annet.device.Storage, NetboxDTO):
    def __init__(self, *args, **kwargs):
       super().__init__(*args, **kwargs)
       self._dev_impl = NetboxAnnetDeviceImpl(self)

    def fqdn(self):
       return self._dev_impl.fqdn()
     ...
```